lightning/src/ln/channelmanager.rs

```diff
@@ -6639,16 +6643,18 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> Writeable f
 			}
 		}
 
-		let channel_state = self.channel_state.lock().unwrap();
-		let mut htlc_purposes: Vec<&events::PaymentPurpose> = Vec::new();
-		(channel_state.claimable_htlcs.len() as u64).write(writer)?;
-		for (payment_hash, (purpose, previous_hops)) in channel_state.claimable_htlcs.iter() {
-			payment_hash.write(writer)?;
-			(previous_hops.len() as u64).write(writer)?;
-			for htlc in previous_hops.iter() {
-				htlc.write(writer)?;
+		let mut htlc_purposes: Vec<events::PaymentPurpose> = Vec::new();
```

Comment:
Hmm I'm not sure if I'm misunderstanding this, but I've adressed this with: 
266175d

But given my understanding, we can't move the non-tlv writer order, so needed complicate the code a bit, take the lock twice and also move the lock order of `claimable_htlcs` to after `pending_outbound_payments`. Let me know if I'm misunderstanding.

Reply:
Ah yeah, but that would violate our idea of not holding locks across different look order branches, no? Perhaps this shows that that entire idea of different lock order branches are a bit flawed...

Reply:
By "lock order branches" I mean the concept we introduced in #1656 with the lock order tree, where for example `forward_htlcs` is in it's own lock order branch. We decided on "Note that locks on different branches shall not be taken at the same time, as doing so will create a new lock order for those specific locks in the order they were taken.".

Though perhaps the issue here is introducing that "claimable_htlcs" shouldn't be in it's own separate branch, as the content of it is stored as is, compared to for example `id_to_peer`.

Reply:
Right ok, thanks for clarifying @TheBlueMatt!
The last part:

> while there is a lockorder introduced, its "fake" in that we can trivially switch it without any impact if we ever want a lockorder later, and we're not holding locks too long here because we're blocking the world anyway.

Clarifies it for me. I'll keep the lock order tree illustration as is then (ie. `claimable_htlcs` as a separate branch), but introduce lock orders only in the serialization that only violates the branch concept, but that we can easily be taken out later.